gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/public/src/components/selectionTree/js/SelectionTree.js

```diff
@@ -0,0 +1,442 @@
+            "visual-alternatives": ".gpiic-oauth2-prefSelection-visual-alternatives",
+            "visual-alternatives.speak-text": ".gpiic-oauth2-prefSelection-visual-alternatives_speak-text",
+            "visual-alternatives.speak-text.rate": ".gpiic-oauth2-prefSelection-visual-alternatives_speak-text_rate",
+            "visual-alternatives.speak-text.volume": ".gpiic-oauth2-prefSelection-visual-alternatives_speak-text_volume",
+            "visual-alternatives.speak-text.pitch": ".gpiic-oauth2-prefSelection-visual-alternatives_speak-text_pitch",
+            "visual-alternatives.speak-text.language": ".gpiic-oauth2-prefSelection-visual-alternatives_speak-text_language",
+            "visual-alternatives.speak-text.announce": ".gpiic-oauth2-prefSelection-visual-alternatives_speak-text_announce",
+            "visual-alternatives.speak-text.read-back": ".gpiic-oauth2-prefSelection-visual-alternatives_speak-text_read-back",
+            "visual-alternatives.speak-text.text-highlighting": ".gpiic-oauth2-prefSelection-visual-alternatives_speak-text_text-highlighting",
+            "visual-alternatives.speak-text.follows": ".gpiic-oauth2-prefSelection-visual-alternatives_speak-text_follows",
+            "visual-alternatives.braille": ".gpiic-oauth2-prefSelection-visual-alternatives_braille",
+            "visual-styling": ".gpiic-oauth2-prefSelection-visual-styling",
+            "visual-styling.change-contrast": ".gpiic-oauth2-prefSelection-visual-styling_change-contrast",
+            "visual-styling.emphasize-links": ".gpiic-oauth2-prefSelection-visual-styling_emphasize-links",
+            "visual-styling.text-style": ".gpiic-oauth2-prefSelection-visual-styling_text-style"
+        },
+        invokers: {
+            updateDOMState: {
+                funcName: "gpii.oauth2.selectionTree.setCheckbox"
+            },
+            bindDOMListener: {
+                funcName: "gpii.oauth2.selectionTree.bindDOMListener",
+                args: ["{that}", "{arguments}.0", "{arguments}.1", "{that}.updateModel"]
+            }
+        }
+    });
+
+    gpii.oauth2.selectionTree.fetchResources = function (that) {
+        fluid.fetchResources(that.options.resources, function () {
+            that.refresh();
+        });
+    };
+
+    gpii.oauth2.selectionTree.refresh = function (that) {
+        that.container.html(that.options.resources.template.resourceText);
+        that.events.afterTemplateLoaded.fire(that);
+    };
+
+    gpii.oauth2.selectionTree.addAria = function (that) {
+        that.locate("tree").attr("role", "tree");
+        that.locate("leaf").attr("role", "treeitem");
+        that.locate("branch").attr("role", "group");
+        that.locate("branchToggle").attr("aria-expanded", false);
+    };
+
+    gpii.oauth2.selectionTree.toggleBranch = function (that, elm, state) {
+        elm = $(elm);
+        var collapseState;
+        var expandState;
+
+        if (arguments.length === 3) {
```

Comment:
I would prefer a check against state === undefined to be less framework-hostile : P